test/unit/storage_sync_test.rb

```diff
@@ -38,9 +40,15 @@ def test_redis_malformed_url
     end
   end
 
+  def test_redis_unknown_scheme
+    assert_raise ArgumentError do
+      StorageSync.send :new, url('myscheme://127.0.0.1:6379')
+    end
+  end
+
   def test_redis_url_without_scheme
     assert_nothing_raised do
-      StorageSync.send :new, url('foo')
+      StorageSync.send :new, url('backend-redis:6379')
```

Comment:
```suggestion
      storage = StorageSync.send :new, url('backend-redis')
      assert_client_config(storage, url: URI('redis://backend-redis:6379'))
```

Just a suggestion... not sure if a good one (obviously, `assert_nothing_raised` is not needed in this case)

I was just surprised that the port was deleted, I think this actually modifies the test. And also, there is another test, `test_redis_host_and_port` which already tests host+port.

Reply:
I applied this: https://github.com/3scale/apisonator/pull/384/commits/de82ec273a4888f7f5d0caf7ab96786d7cac90c1